tools/psa/build_tfm.py

```diff
@@ -160,22 +182,334 @@ def clean_up_cloned_repos():
+    """
+    # Use --intent-to-add option of git status to identify un-tracked files
+    cmd = ['git', '-C', ROOT, 'status', 'N', directory]
+    run_cmd_and_return_errorcode(cmd)
+
+    cmd = ['git', '-C', ROOT, 'diff', '--exit-code', '--quiet', directory]
+    changes_made = run_cmd_and_return_errorcode(cmd)
+
+    if changes_made:
+        if target:
+            logger.info("Change in image for %s has been detected" % target)
+        else:
+            logger.info("Change detected in directory %s" % directory)
+        cmd = ['git', '-C', ROOT, 'add', relpath(directory, ROOT)]
+        run_cmd_and_return_errorcode(cmd)
+
+        if target:
+            logger.info("Committing image for %s" % target)
+            msg = '--message="Updated secure binary for %s (%s)"' % (target,
+                                                                     toolchain)
+        else:
+            logger.info("Committing changes in directory %s" % directory)
+            msg = '--message="Updated directory %s "' % directory
+
+        cmd = ['git', '-C', ROOT, 'commit', msg]
+        run_cmd_and_return_errorcode(cmd)
+    else:
+        if target:
+            logger.info("No changes detected in %s, skipping commit" % target)
+        else:
+            logger.info("No changes detected in %s, skipping commit" %
+                                                    relpath(directory, ROOT))
+
+def run_cmake_build(configure, cmake_build_dir, bl_supported= None,
+                    tfm_target=None, tfm_toolchain=None, debug=False):
+
+    if configure:
+        cmake_cmd = ['cmake', '-GUnix Makefiles']
+        cmake_cmd.append('-DPROJ_CONFIG=' + (join(TF_M_BUILD_DIR,
+                         'trusted-firmware-m', 'configs/ConfigCoreIPC.cmake')))
+        cmake_cmd.append('-DTARGET_PLATFORM=' + tfm_target)
+        cmake_cmd.append('-DCOMPILER=' + tfm_toolchain)
+        if debug:
+            cmake_cmd.append('-DCMAKE_BUILD_TYPE=Debug')
+        else:
+            cmake_cmd.append('-DCMAKE_BUILD_TYPE=Release')
+        if not bl_supported:
+            cmake_cmd.append('-DBL2=False')
+        cmake_cmd.append('..')
+    else:
+        # install option exports NS APIs to a dedicated folder under
+        # cmake build folder
+        cmake_cmd = ['cmake', '--build', '.', '--', 'install']
+
+    proc = subprocess.Popen(cmake_cmd, stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE, cwd=cmake_build_dir)
+    for line in iter(proc.stdout.readline, b''):
+        logger.debug(line.decode("utf-8").strip('\n'))
+
+    std_out, std_err = proc.communicate()
+
+    if proc.returncode:
+        logger.info(std_err.decode("utf-8"))
+    return proc.returncode
+
+def build_tfm(args):
+    clone_tfm_repo(args.commit)
+
+    cmake_build_dir = join(TF_M_BUILD_DIR, 'trusted-firmware-m',
+                            'cmake_build')
+    if not isdir(cmake_build_dir):
+        os.mkdir(cmake_build_dir)
+
+    debug = False
+    if args.mcu:
```

Comment:
The if and else statement seems to have Code duplication, we could just modularise or put it in a single loop

Reply:
Fixed